scripts/record_and_replay.sh

```diff
@@ -0,0 +1,192 @@
+
+join_by() {
+    joined=$(printf ",'%s'" "${arr[@]}")
+    IFS="$1"
+    shift
+    echo "$*"
+}
+
+while (("$#")); do
+    case "$1" in
+    --source)
+        source_workspace=$2
+        shift 2
+        ;;
+    --target)
+        target_workspace=$2
+        shift 2
+        ;;
+    --tf-dir)
+        tf_dir=$2
+        shift 2
+        ;;
+    --var-file)
+        var_file=$2
+        shift 2
+        ;;
+    --dry-run)
+        echo "Executing dry run. No changes to Terraform will be made."
+        dry_run=1
+        shift 1
+        ;;
+    --verbose)
+        verbose=1
+        shift 1
+        ;;
+    *)
+        break
+        ;;
+    esac
+done
+
+if [ -z "$source_workspace" ] || [ -z "$target_workspace" ] || [ -z "$var_file" ]; then
+    echo "Missing argument"
+    echo "Usage:"
+    echo "record_and_replay --source <source workspace> --target <target workspace> --var-file <terraform tfvars file> [--dry-run] [--verbose]"
+    echo
+    echo "Examples:"
+    echo "record_and_replay --source dev --target devtest --var-file dev.tfvars # replay dev network to new devtest workspace"
+    exit 1
+fi
+
+# get to terraform dir
+SCRIPT_PATH="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+if [ -z "$tf_dir" ]; then
+    cd "$SCRIPT_PATH/../terraform"
+else
+    cd $tf_dir || exit 1
+fi
+
+echo -e "Searching workspaces in $PWD...\n"
+
+terraform workspace select $source_workspace >/dev/null 2>&1
+if [ $? -eq 0 ]; then
+    echo "Using source workspace \"$source_workspace\""
+else
+    echo "ERROR: Could not find workspace \"$source_workspace\". Is it initialized?"
+    exit 1
+fi
+
+terraform workspace select $target_workspace >/dev/null 2>&1
+if [ $? -eq 0 ]; then
+    echo "Using target workspace \"$target_workspace\""
+else
+    echo "Could not find workspace \"$target_workspace\". Would you like to initialize a new workspace?"
+    echo -e "Only \"yes\" will be accepted to confirm.\n"
+    echo "Enter a value: "
+    read confirmation
+    if [ "$confirmation" = "yes" ]; then
+        if [ -z "$dry_run" ]; then
+            echo "EXECUTE: terraform workspace new $target_workspace"
+            terraform workspace new $target_workspace ｜｜ exit 1
+        else
+            echo "DRY-EXECUTE: terraform workspace new $target_workspace"
+            echo
+        fi
+    else
+        echo "Record and replay cancelled"
+        exit 1
+    fi
+fi
+
+# done fetching terraform workspaces
+# now get aws data
+echo
+
+######
+# Fetch validator info (restore EBS volumes and private IPs) from aws and transform them into terraform vars
+# NOTE: overrides IFS to maintain tab and newline formatting from aws cli
+######
+
+OLDIFS=$IFS
+
+IFS="" # set IFS to retain formatting of aws cli output
+validators_description="$(aws ec2 describe-instances --filter "Name=tag:Workspace,Values=$source_workspace" "Name=tag:Role,Values=validator" --query "Reservations[*].Instances[*].[BlockDeviceMappings,PrivateIpAddress,Tags[*]]" --output text)"
```

Comment:
You could also get this information from `terraform state show 'aws_instance.validator[0]'`.